### PR TITLE
Hours 120 rails 7

### PR DESCRIPTION
**Pull request by barmintor**
This is a WIP branch for upgrading to Rails 7. It includes an interim update to Rails 6, which permits a refactor of the `UPSERT` logic to work against sqlite (which will make it easier to develop)

### CODE REVIEW

**Comment by barmintor** [comment]
@edmadrid @er-k @elohanlon this PR isn't ready yet, but FYI

File: Gemfile [hunk 1]
```diff
@@ -67,11 +69,11 @@ end
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 # Moved outside dev/test group to get deploy to work, known issue
-gem 'listen', '>= 3.0.5', '< 3.2'
+gem 'listen', '~> 3.2'
 
 # Login/auth gems
 gem 'devise'
-gem 'cul_omniauth', '~> 0.7.0'
+gem 'cul_omniauth', github: "cul/cul_omniauth", branch: "ldpd-357_rails-7"
```
Comment (elohanlon) [inline Gemfile]: I know you said you're going to update this after the gem release, so this comment is just a reminder about that.

File: app/models/timetable.rb [hunk 2]
```diff
@@ -44,20 +43,17 @@ def self.batch_update_or_create(timetable_params, open, close)
         open_time = nil
         close_time = nil
       end
-      values.push(timetable_params['location_id'], day, open_time, close_time, timetable_params['closed'], timetable_params['tbd'], timetable_params['note'], Time.current, Time.current)
+      attr_values = [timetable_params['location_id'], day, open_time, close_time, timetable_params['closed'], timetable_params['tbd'], timetable_params['note'], Time.current, Time.current]
+      attr_names.zip(attr_values).to_h
```
Comment (elohanlon) [inline app/models/timetable.rb]: Using `zip` feels a little more fragile to me than defining the key-value pairs inside map.  If someone reorders those attr_names or adds a new value, seems easy to mismatch things in attr_values.

Might be less error-prone (in the case of future changes) to do something like this?
```
attrs = {
  location_id: timetable_params['location_id'],
  date: day,
  ...
  ...
  ...
}
```

File: Gemfile (hunk 1 above)
Comment (barmintor) [reply Gemfile]: thanks! Just pushed those updates up.

**Review by elohanlon — APPROVED** [review]
Approving PR, with one optional recommendation about the use of `zip`!

File: app/models/timetable.rb (hunk 2 above)
Comment (barmintor) [reply app/models/timetable.rb]: I'm amenable to this.

**Pull request merged**